src/NSwag.SwaggerGeneration.AspNetCore/OperationResponseProcessor.cs

```diff
@@ -0,0 +1,222 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="OperationResponseProcessor.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using NJsonSchema;
+using NJsonSchema.Generation;
+using NJsonSchema.Infrastructure;
+using NSwag.SwaggerGeneration.Processors;
+using NSwag.SwaggerGeneration.Processors.Contexts;
+using NSwag.SwaggerGeneration.WebApi.Processors.Models;
+
+namespace NSwag.SwaggerGeneration.AspNetCore
+{
+    /// <summary>Generates the operation's response objects based on reflection and the ResponseTypeAttribute, SwaggerResponseAttribute and ProducesResponseTypeAttribute attributes.</summary>
+    public class OperationResponseProcessor : IOperationProcessor
+    {
+        private readonly AspNetCoreToSwaggerGeneratorSettings _settings;
+
+        /// <summary>Initializes a new instance of the <see cref="OperationParameterProcessor"/> class.</summary>
+        /// <param name="settings">The settings.</param>
+        public OperationResponseProcessor(AspNetCoreToSwaggerGeneratorSettings settings)
+        {
+            _settings = settings;
+        }
+
+        /// <summary>Processes the specified method information.</summary>
+        /// <param name="operationProcessorContext"></param>
+        /// <returns>true if the operation should be added to the Swagger specification.</returns>
+        public async Task<bool> ProcessAsync(OperationProcessorContext operationProcessorContext)
+        {
+            if (!(operationProcessorContext is AspNetCoreOperationProcessorContext context))
+                return false;
+
+            var parameter = context.MethodInfo.ReturnParameter;
+            var successXmlDescription = await parameter.GetDescriptionAsync(parameter.GetCustomAttributes()).ConfigureAwait(false) ?? string.Empty;
+
+            var responseTypeAttributes = context.MethodInfo.GetCustomAttributes()
+                .Where(a => a.GetType().Name == "ResponseTypeAttribute" ||
+                            a.GetType().Name == "SwaggerResponseAttribute")
+                .Concat(context.MethodInfo.DeclaringType.GetTypeInfo().GetCustomAttributes()
+                    .Where(a => a.GetType().Name == "SwaggerResponseAttribute"))
+                .ToList();
+
+            if (context.ApiDescription.SupportedResponseTypes.Any())
```

Comment:
I believe ApiDescription will also add a default `200` response type no matter what. This all looks like it needs more testing and thought

Reply:
Currently one rule is important: If at least one SwaggerResponeAttribute or SwaggerDefaultResponseAttribute is defined than only these are used for the Swagger/OpenApi spec. More https://github.com/RSuter/NSwag/wiki/WebApiToSwaggerGenerator#additionally-supported-attributes

Reply:
Yeah, I suppose my comment was in reference to the `.Any()` - I'm not sure it's protecting anything.

I thing having the swagger/nswag attributes (when present) override the framework's description is the right thing to do in this case 👍